src/librados/AioCompletionImpl.h

```diff
@@ -85,10 +85,7 @@ struct librados::AioCompletionImpl {
     return wait_for_complete();
   }
   int is_complete() {
-    lock.Lock();
-    int r = complete;
-    lock.Unlock();
-    return r;
+    return complete;
```

Comment:
right, on amd64, stores are not reordered with older stores. i didn't check older x86 archs. and amd and intel are using MESI protocols to ensure the cache coherence.

but please note that the compiler has the luxury to perform compile-time reordering. without the protection of the mutex. if the caller of `is_complete()` could run into a `rval` not yet set.